lib/SecretsCrypto/secrets.c

```diff
@@ -0,0 +1,1000 @@
+		    Sailfish_Secrets_Callback_Data_new(
+			    (Sailfish_Secrets_SecretManager_result_callback)callback,
+			    NULL,
+			    callback_context);
+		daemon_connection.busy = 1;
+		g_dbus_proxy_call(daemon_connection.secretsProxy,
+				  "setSecret",
+				  g_variant_new("(@((ss)aya{sv})@(i)s)",
+						Sailfish_Secrets_variantFromSecret(secret),
+						g_variant_new("(i)", uiMode),
+						interactionServiceAddress),
+				  G_DBUS_CALL_FLAGS_NONE, -1, NULL,
+				  Sailfish_Secrets_setSecretReady, cbd);
+		return 1;
+	}
+}
+
+int Sailfish_Secrets_SecretManager_getSecret(
+		struct Sailfish_Secrets_Secret_Identifier *ident,
+		enum Sailfish_Secrets_SecretManager_UserInteractionMode uiMode,
+		const char *interactionServiceAddress,
+		Sailfish_Secrets_SecretManager_getSecret_callback callback,
+		void *callback_context)
+{
+	if (Sailfish_Secrets_busy() || !Sailfish_Secrets_connectedToServer()) {
+		return 0;
+	} else {
+		struct Sailfish_Secrets_Callback_Data *cbd =
+		    Sailfish_Secrets_Callback_Data_new(
+			    NULL,
+			    (Sailfish_Secrets_SecretManager_secret_result_callback) callback,
+			    callback_context);
+		daemon_connection.busy = 1;
+		g_dbus_proxy_call(daemon_connection.secretsProxy,
+				  "getSecret",
+				  g_variant_new("(@(ss)@(i)s)",
+						Sailfish_Secrets_variantFromSecretIdentifier(ident),
+						g_variant_new("(i)", uiMode),
+						interactionServiceAddress),
+				  G_DBUS_CALL_FLAGS_NONE, -1, NULL,
+				  Sailfish_Secrets_getSecretReady, cbd);
+		return 1;
+	}
+}
+
+int Sailfish_Secrets_SecretManager_deleteSecret(
+		struct Sailfish_Secrets_Secret_Identifier *ident,
+		enum Sailfish_Secrets_SecretManager_UserInteractionMode uiMode,
+		const char *interactionServiceAddress,
+		Sailfish_Secrets_SecretManager_deleteSecret_callback callback,
+		void *callback_context)
+{
+	if (Sailfish_Secrets_busy() || !Sailfish_Secrets_connectedToServer()) {
+		return 0;
+	} else {
+		struct Sailfish_Secrets_Callback_Data *cbd =
+		    Sailfish_Secrets_Callback_Data_new(
+			    (Sailfish_Secrets_SecretManager_result_callback) callback,
+			    NULL,
+			    callback_context);
+		daemon_connection.busy = 1;
+
+		g_dbus_proxy_call(daemon_connection.secretsProxy,
+				  "deleteSecret",
+				  g_variant_new("(@(ss)@(i)s)",
+						Sailfish_Secrets_variantFromSecretIdentifier(ident),
+						g_variant_new("(i)", uiMode),
+						interactionServiceAddress),
+				  G_DBUS_CALL_FLAGS_NONE, -1, NULL,
+				  Sailfish_Secrets_deleteSecretReady, cbd);
+		return 1;
+	}
+}
+
+/****************************** Daemon Connection *******************/
+
+int Sailfish_Secrets_busy()
+{
+	return daemon_connection.busy;
+}
+
+int Sailfish_Secrets_connectedToServer()
+{
+	return daemon_connection.secretsProxy != NULL;
+}
+
+int Sailfish_Secrets_connectToServer(
+		Sailfish_Secrets_connectToServer_callback callback,
+		void *callback_context)
+{
+	if (Sailfish_Secrets_connectedToServer()) {
+		return 2;	/* already connected */
```

Comment:
A magic constant

Reply:
Will add to the private constants header when I add that one